brushfire-tree/src/main/scala/com/stripe/brushfire/Error.scala

```diff
@@ -1,7 +1,20 @@
 package com.stripe.brushfire
 
-import com.twitter.algebird._
-
+import com.twitter.algebird.{AveragedValue, Group, Monoid, Semigroup}
+
+/** Computes some kind of error by comparing the trees' predictions to the validation set */
+trait Error[T, P, E] {
```

Comment:
I think you probably have this in here because pruning requires it, but pruning is really a training thing, and we should move this to training and then extract pruning from tree. (Or, IMO it even would be fine to temporarily lose the ability to prune and file an issue to bring it back).